Woof/Woof/ViewModel/SitterProfileViewModel.swift

```diff
@@ -91,35 +84,55 @@ final class SitterProfileViewModel: ObservableObject {
 
     private func upload(_ sitter: Sitter) async throws {
         let endpoint = WoofAppEndpoint.addNewSitter(sitter.asDictionary())
-        _ = try await NetworkService<WoofAppEndpoint>().request(endpoint)
+
+        do {
+            _ = try await NetworkService<WoofAppEndpoint>().request(endpoint)
+        } catch {
+            throw AppError.uploadFailed
+        }
     }
 
-    private func saveLocally(_ sitter: Sitter) async {
+    private func saveLocally(_ sitter: Sitter) async throws {
         guard let data = try? JSONEncoder().encode(sitter),
               KeyValueStorage(KeyValueStorage.Name.currentSitter)
               .save(data, for: KeyValueStorage.Key.currentSitter) else {
-            await handleError(.saveLocallyFailed)
+            throw AppError.saveLocallyFailed
+        }
+    }
+
+    @MainActor private func handleError(_ error: Error) {
+        guard let appError = error as? AppError else {
+            errorMessage = "An error occurred."
             return
         }
+
+        errorMessage = appError.errorDescription
     }
 
-    @MainActor private func handleError(_ error: AppError) {
-        errorMessage = error.errorDescription
+    /// Indicates whether there is a saved sitter.
+    private(set) var sitterIsSet = false
+    private var currentSitter: Sitter? {
+        didSet {
+            sitterIsSet = true
+        }
     }
 
-    private func loadSitterFromStorage() -> Sitter {
+    private func loadSitterFromStorage() -> Sitter? {
```

Comment:
Could you organize private interfaces carefully, please don't mix properties and methods. It looks really messy when you see properties, then some methods, then again some properties, and again some methods.

Reply:
Fixed with e9d20b2